File: 0942-di-string-match/0942-di-string-match.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        int n=s.size();
        vector<int> v;
        int i=0;
        int j=n;
        for(char c:s){
            if(c=='I'){
                v.push_back(i++);
            }
            else{
                v.push_back(j--);
            }
        }
        v.push_back(i);
        return v;
    }
};

File: 0942-di-string-match/0942-di-string-match.js
/**
 * @param {string} s
 * @return {number[]}
 */
var diStringMatch = function(s) {
    let n=s.length;
    let arr=new Array(n+1);
    let left=0;
    let right=n;
    let i=0;
    for(let c of s){
        if(c=='I'){
            arr[i]=left;
            left++;
        }
        else{
            arr[i]=right;
            right--;
        }
        i++;
    }
    arr[i]=left;
    return arr;
};